plugins/CoreHome/Controller.php

```diff
@@ -86,7 +86,33 @@ public function renderWidget($widget)
     {
         Piwik::checkUserHasSomeViewAccess();
 
-        return $widget->render();
+        $config = new WidgetConfig();
+        $widget::configure($config);
+
+        $content = $widget->render();
+
+        if ($config->getName() && Common::getRequestVar('showtitle', '', 'string') === '1') {
+            if (strpos($content, '<h2') !== false
+                || strpos($content, ' content-title=') !== false
+                || strpos($content, ' piwik-enriched-headline') !== false
+                || strpos($content, '<h1') !== false ) {
+                // already includes title
+                return $content;
+            }
+
+            if (strpos($content, 'piwik-content-block') === false
+                && strpos($content, 'class="card"') === false
```

Comment:
We might not detect in all cases whether someone is using the card css class manually but developers should not do this anyway, they should use piwik-content-block. So it's fine if someone was using `class=" card test"`